### PR TITLE
Update modulefiles and UFS Weather model build option

### LINKED ISSUE #4: Update modulefiles to combine with UFS-WM NOAH-MP subcomponent
- Add dual functionality to build UFS-WM -DAPP=ATML application in the ecbuild-based land-DA_workflow cmake system
- Set UFS-WM as submodule and utilize compile.sh and run_compile.sh scripts
- Additional build scripts can be added to CMakeLists.txt
- Depends on unified spack-stack maintenance on RDHPCS and PW-Cloud platforms: need to prioritize hera/orion cases first
- Follow-up is needed with https://github.com/ufs-community/ufs-weather-model/issues/1651

### PR DESCRIPTION
**Pull request by jkbk2004**
## Description
- Updated modulefiles (hera/orion) required to build UFS Weather model
- New modulefiles are synced with https://github.com/ufs-community/ufs-weather-model/pull/1707
- Updated CMakeList to add UFS Weather model build with external project 

## Subcomponents involved:
- [ ] DA_update
- [ ] vector2tile
- [ ] ufs-land-driver
- [x] none

## Linked PR's and Issues:
Closes https://github.com/ufs-community/land-DA_workflow/issues/4

### Testing (for CM's):
- RDHPCS
    - [*] Hera
    - [ ] Orion
    - [ ] Jet
    - [ ] Gaea
    - [ ] Cheyenne
- CI
  - [ ] Completed
- PW-louds
  - [ ] AWS
  - [ ] AZURE
  - [ ] GCP

- Need to update container according to the updated modulefiles
- Tips for build test steps:
  - module use modulefiles -> module load ufs_hera.intel or ufs_orion.intel -> module load ufs_common
  - mkdir build -> cd build -> ecbuild .. -> make -j 4

### CODE REVIEW

**Comment by jkbk2004** [comment]
@ulmononian FYI: we need to update container to fully follow up with the modulefile updates.

**Comment by ulmononian** [comment]
> @ulmononian FYI: we need to update container to fully follow up with the modulefile updates.

i believe @climbfuji is working on an official spack-stack/1.3.1 container for the ufs-wm. further, @mark-a-potts has created a container to run the WM RTs. however, i am not sure which container is/will be best suited for this case. any thoughts?

**Comment by climbfuji** [comment]
Just to confirm that I am working on it, yes.

**Review by rhaesung — DISMISSED** [review]

**Review by ulmononian — DISMISSED** [review]
like butter! @jkbk2004

**Review by yichengt90 — DISMISSED** [review]
Successfully build ufs-weather-model with externalProject build.

**Comment by jkbk2004** [comment]
@barlage FYI: we are adding ufs-weather model build option in the ecbuild-based cmake for the use of Noah-MP -DAPP=LND. We can start using Noah-MP weather model baseline as our starting point.

File: modulefiles/ufs_orion.intel.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+load("ufs_common")
+
+setenv("CC", "mpiicc")
+setenv("CXX", "mpiicpc")
+setenv("FC", "mpiifort")
+setenv("CMAKE_Platform", "orion.intel")
+
+setenv("EPICHOME", "/work/noaa/epic-ps/role-epic-ps")
+setenv("JEDI_INSTALL", pathJoin(os.getenv("EPICHOME"),"contrib"))
```
Comment (ulmononian) [inline modulefiles/ufs_orion.intel.lua]: we should ensure this points to the fv3-bundle built with the spack-stack ue (that path is `work/noaa/epic-ps/role-epic-ps/contrib/fv3-bundle-UE`). currently there are two bundles there (fv3-bundle and fv3-bundle-UE), but fv3-bundle  won't work if we use the new stack.

File: modulefiles/ufs_hera.intel.lua [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+mapl_ver=os.getenv("mapl_ver") or "2.22.0-esmf-8.3.0b09"
+load(pathJoin("mapl", mapl_ver))
+
+setenv("CC", "mpiicc")
+setenv("CXX", "mpiicpc")
+setenv("FC", "mpiifort")
+setenv("CMAKE_Platform", "hera.intel")
+
+setenv("EPICHOME", "/scratch1/NCEPDEV/nems/role.epic")
+setenv("JEDI_INSTALL", pathJoin(os.getenv("EPICHOME"),"contrib"))
```
Comment (ulmononian) [inline modulefiles/ufs_hera.intel.lua]: same comment as for orion. hera path is `/scratch1/NCEPDEV/nems/role.epic/contrib/fv3-bundle-UE`.

**Comment by jkbk2004** [comment]
@zach1221 I tested ok with singularity. can you merge in the pr?

**Review by zach1221 — APPROVED** [review]

**Comment by zach1221** [comment]
@jkbk2004 yes I'll merge

**Comment by zach1221** [comment]
@jkbk2004 are you able to mark the two pending conversations as resolved?

**Pull request merged**